The BLKFLSBUF and FDFLUSH ioctls are Linux specific, and do not
really have anything to do with __GNUC__ (which is also used on
OS/X and Solaris).  No need to print these warnings on OS/X.

Signed-off-by: Andreas Dilger <adilger@sun.com>

Index: e2fsprogs-1.41.10/lib/ext2fs/flushb.c
===================================================================
--- e2fsprogs-1.41.10.orig/lib/ext2fs/flushb.c
+++ e2fsprogs-1.41.10/lib/ext2fs/flushb.c
@@ -65,17 +65,13 @@ errcode_t ext2fs_sync_device(int fd, int
 #ifdef BLKFLSBUF
 		if (ioctl (fd, BLKFLSBUF, 0) == 0)
 			return 0;
-#else
-#ifdef __GNUC__
- #warning BLKFLSBUF not defined
-#endif /* __GNUC__ */
+#elif defined(__linux__)
+#warning BLKFLSBUF not defined
 #endif
 #ifdef FDFLUSH
 		ioctl (fd, FDFLUSH, 0);   /* In case this is a floppy */
-#else
-#ifdef __GNUC__
- #warning FDFLUSH not defined
-#endif /* __GNUC__ */
+#elif defined(__linux__)
+#warning FDFLUSH not defined
 #endif
 	}
 	return 0;
